nautobot/extras/views.py

```diff
@@ -2064,56 +2064,67 @@ class JobHookUIViewSet(NautobotUIViewSet):
-            ),
-            success_log_count=count_related(
-                JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_SUCCESS}
-            ),
-            info_log_count=count_related(
-                JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_INFO}
-            ),
-            warning_log_count=count_related(
-                JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_WARNING}
-            ),
-            error_log_count=count_related(
-                JobLogEntry,
-                "job_result",
-                filter_dict={
-                    "log_level__in": [
-                        LogLevelChoices.LOG_FAILURE,
-                        LogLevelChoices.LOG_ERROR,
-                        LogLevelChoices.LOG_CRITICAL,
-                    ],
-                },
-            ),
+def get_annotated_jobresult_queryset(queryset=None):
+    if get_settings_or_config("JOB_RESULTS_SUMMARY_FIELD_ENABLED") and queryset:
+        return (
+            queryset.defer("result")
+            .select_related("job_model", "user")
+            .annotate(
+                debug_log_count=count_related(
+                    JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_DEBUG}
+                ),
+                success_log_count=count_related(
+                    JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_SUCCESS}
+                ),
+                info_log_count=count_related(
+                    JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_INFO}
+                ),
+                warning_log_count=count_related(
+                    JobLogEntry, "job_result", filter_dict={"log_level": LogLevelChoices.LOG_WARNING}
+                ),
+                error_log_count=count_related(
+                    JobLogEntry,
+                    "job_result",
+                    filter_dict={
+                        "log_level__in": [
+                            LogLevelChoices.LOG_FAILURE,
+                            LogLevelChoices.LOG_ERROR,
+                            LogLevelChoices.LOG_CRITICAL,
+                        ],
+                    },
+                ),
+            )
         )
-    )
+    else:
+        return JobResult.objects.defer("result").select_related("job_model", "user")
 
 
 class JobResultListView(generic.ObjectListView):
     """
     List JobResults
     """
 
-    queryset = get_annotated_jobresult_queryset()
+    queryset = JobResult.objects.all()
     filterset = filters.JobResultFilterSet
     filterset_form = forms.JobResultFilterForm
     table = tables.JobResultTable
     action_buttons = ()
 
+    def alter_queryset(self, request):
+        """
+        Override to use the annotated queryset
+        """
+        queryset = super().alter_queryset(request)
+
+        return get_annotated_jobresult_queryset(queryset)
+
```

Comment:
Shouldn't there be something either in the view or in the table to hide the `summary` column altogether if the setting is disabled? I'm thinking maybe in the view's `get_extra_context()`?

Reply:
Doesn't that have the problem that it's defined at declaration time, and so won't change if the Constance setting gets changed?